bftengine/tests/controllerWithSimpleHistory/ControllerWithSimpleHistory_test.cpp

```diff
@@ -0,0 +1,133 @@
+// Concord
+//
+// Copyright (c) 2020 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the "License"). You may not use this product except in
+// compliance with the Apache 2.0 License.
+//
+// This product may include a number of subcomponents with separate copyright notices and license terms. Your use of
+// these subcomponents is subject to the terms and conditions of the sub-component's license, as noted in the LICENSE
+// file.
+
+#include "ControllerWithSimpleHistory.hpp"
+#include "gtest/gtest.h"
+#include <chrono>
+#include <thread>
+
+using namespace std;
+using namespace bftEngine;
+
+// Tests methods - insideActiveWindow, onBecomePrimary.
+// Tests logic, which sets the range of sequence numbers that are valid for current window.
+// The starting index is initialized to the next sequence number:
+// that is a multiplication of EvaluationPeriod plus one.
+// The end index is `start + EvaluationPeriod` excluded.
+TEST(ControllerWithSimpleHistory, sequence_number_range_window) {
+  // Test case - init seq is 0.
+  // range should be (1,1+ontrollerWithSimpleHistory::EvaluationPeriod]
+  {
+    // Replica, id = 0 , is primary, view 0, 0 seq num.
```

Comment:
Perhaps, rather than a comment, use named variables for these magic numbers. That will make the code more legible in my opinion.